### PR TITLE
Complete authentication in React

### PR DESCRIPTION
**Pull request by shannon-n-lin**
- Fixed validation for log in and sign up pages, added field-specific error messages
- Sign up form will now automatically log in new users and redirect to index
- Created user context so user data persists across components after log in
- Added conditional formatting to desktop and mobile nav menus so authenticated users do not see log in or sign up buttons, guests do not see log out button

### CODE REVIEW

File: .vscode/settings.json
Comment (okasen) [inline .vscode/settings.json:1]: For future PRs, you can prevent files like this from appearing in commits with a gitignore-- there should even be some templates for vscode gitignores that will automatically catch files generated by vscode :)

File: backend/server.js [hunk 1]
```diff
@@ -27,7 +27,11 @@ app.use(express.urlencoded({ extended: true }))
 app.use(express.json())
 
 // CORS enabling middleware
-app.use(cors());
+app.use(cors({
+  origin: 'http://localhost:5173',
```
Comment (okasen) [inline backend/server.js:31]: Might be worth keeping the origin url in an environment variable, but that can be a later improvement depending on how you're hosting this (I have forgotten!)

File: frontend/src/App.jsx [hunk 2]
```diff
@@ -1,22 +1,48 @@
...
 import LoginPage from './pages/LoginPage'
 import SignupPage from './pages/SignupPage'
 
 
 const App = () => {
+  const [user, setUser] = useState('')
+
+  useEffect(() => {
+    const checkAuth = async () => {
+      const res = await axios.get('http://localhost:2003/user', {
```
Comment (okasen) [inline frontend/src/App.jsx:16]: Clarifying my comment about environment variables from above (since it applies here too):

If you have the domain you're using (localhost, sunscreensavvy.com or anything else) as an environment variable, then you can swap them out depending on the environment you have your code deployed to-- perhaps on your local machine it's just localhost, and the production version of the site is on the domain-named url.

So basically instead of "get('localhost:2003/user')" it'd be something like get("{BACKEND_ROUTE}/user" where BACKEND_ROUTE evaluates to your url.

 (This might not be too useful yet, but might be good to keep in mind for the future)

**Review by okasen — APPROVED** [review]
This looks really great! I left some comments for future iterations, feel free to ask any questions on them, but approving this now!
